### PR TITLE
fixed java.lang.StringIndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by hone-liu**
fixed java.lang.StringIndexOutOfBoundsException where value is an empty string

### CODE REVIEW

**Comment by gsmet** [comment]
Hi @hone-liu ,

Thanks for this PR. Do you have a reproducible test case by any chance? That would help avoiding regression in the future.

To be honest, I'm a bit surprised you can send an empty string as a valid value for a char.

**Comment by JWGmeligMeyling** [comment]
Not sure if this change is the right way to achieve this. @vladmihalcea has released a custom type for this: https://vladmihalcea.com/how-to-implement-a-custom-basic-type-using-hibernate-usertype/ . Maybe he can shed some light on this.

File: hibernate-core/src/main/java/org/hibernate/type/descriptor/java/CharacterTypeDescriptor.java [hunk 1]
```diff
@@ -58,6 +58,9 @@ public Character fromString(String string) {
 		}
 		if ( String.class.isInstance( value ) ) {
 			final String str = (String) value;
+			if(str.length() == 0){
+				return null;
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/type/descriptor/java/CharacterTypeDescriptor.java]: Shouldn't we use `Character.MIN_VALUE` instead?

**Comment by gsmet** [comment]
Yes, I must admit I'm not very convinced. That's why I would like to see a proper use case.

**Comment by beikov** [comment]
Closing due to the lack of a JIRA issue and test case

**Pull request closed without merging**